executor/adapter.go

```diff
@@ -595,6 +595,19 @@ func (a *ExecStmt) handlePessimisticLockError(ctx context.Context, err error) (E
 		if conflictCommitTS > forUpdateTS {
 			newForUpdateTS = conflictCommitTS
 		}
+	} else if terror.ErrorEqual(err, tikv.ErrLockFailNoWait) {
+		// for nowait, when ErrLock happened, ErrLockFailNoWait will be returned, and in the same txn
+		// the select for updateTs must be updated, otherwise there maybe rollback problem.
+		// begin;  select for update key1(here ErrLocked or other errors(or max_execution_time like util),
+		//         key1 lock not get and async rollback key1 is raised)
+		//         select for update key1 again(this time lock succ(maybe lock released by others))
+		//         the async rollback operation rollbacked the lock just acquired
+		newForUpdateTS, tsErr := a.Ctx.GetStore().GetOracle().GetTimestamp(ctx)
+		if tsErr != nil {
+			return nil, tsErr
```

Comment:
if fail happened here,  `UpdateTS` will not be updated, and the previous async rollback is possiblely ongoing, 
still risk that  the later `select for update` rollbacked ? @coocood

Reply:
This error rarely happens, so even if this happened, the async rollback causes the transaction to fail, it is acceptable.

Reply:
@tiancaiamao  @coocood @jackysp 
I'll add and use `getWithRetry`,   but I think here still risk with `nowait`.  User get this statement result error, next statement pessimistic lock `getForUpdateTs` not updated(select for update, update, delete operations),  ths "async rollback" risk still exists,  how should we process the error here?

Reply:
sync rollback may still send duplicated requests later.